packages/thirdweb/src/analytics/track/siwe.ts

```diff
@@ -0,0 +1,69 @@
+import type { Ecosystem } from "src/wallets/in-app/core/wallet/types.js";
+import type { ThirdwebClient } from "../../client/client.js";
+import { stringify } from "../../utils/json.js";
+import { track } from "./index.js";
+
+type SiweEvent = {
+  client: ThirdwebClient;
+  ecosystem?: Ecosystem;
+  walletAddress?: string;
+  walletType?: string;
+  chainId?: number;
+  error?: {
+    message: string;
+    code: string;
+  };
+};
+
+type SiweSuccessEvent = SiweEvent & {
+  error?: undefined;
+};
+
+/**
+ * @internal
+ */
+export async function trackLogin(event: SiweSuccessEvent) {
+  return trackSiweEvent({
+    ...event,
+    type: "login:success",
+  });
+}
+
+type SiweErrorEvent = SiweEvent & {
+  error: {
+    message: string;
+    code: string;
+  };
+};
+
+/**
+ * @internal
+ */
+export async function trackLoginError(event: SiweErrorEvent) {
+  return trackSiweEvent({
+    ...event,
+    type: "login:error",
+  });
+}
+
+/**
+ * @internal
+ */
+async function trackSiweEvent(
+  event: SiweEvent & {
+    type: "login:success" | "login:error";
```

Comment:
Nit: would just call this action so it matches the schema. Wasn't sure what it mapped to when reading the call site

Reply:
Also not sure we need two login actions, I think it's simpler and more efficient to query all 'login' action, and then split by errorCode. 

We should put a default errorCode for error events like "unknown error" in case the err msg is empty